### PR TITLE
lab1 cleaned up

### PR DESCRIPTION
**Pull request by Nowak0**
wydaje mi sie ze executeProgram powinien zostac w main.

### CODE REVIEW

File: lab1PO/FigureList.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		newFigure = new Trojkat(a, b, c);
+		figures[listLength - 1] = newFigure;
+		break;
+	default:
+		break;
+	}
+}
+
+
+void FigureList::updateList(char operation) {
```
Comment (facku8) [inline lab1PO/FigureList.cpp:93]: Split into smaller functions with relevant names.

"updateList" does too much and the name does not tell what.

File: lab1PO/FigureList.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	this->next = object;
+}
+
+
+void FigureList::setListLength(const int length) {
+	this->listLength = length;
+}
+
+
+void FigureList::addNewElement() {
```
Comment (facku8) [inline lab1PO/FigureList.cpp:61]: This could also be splitted in 3 separate functions

File: lab1PO/FigureList.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	this->current = object;
+}
+
+
+void FigureList::setNext(FiguraPlaska* object) {
+	this->next = object;
+}
+
+
+void FigureList::setListLength(const int length) {
```
Comment (facku8) [inline lab1PO/FigureList.cpp:56]: Why is this needed?

File: lab1PO/FigureList.cpp [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	return next;
+}
+
+
+int FigureList::getListLength() {
+	return listLength;
+}
+
+
+void FigureList::setPrevious(FiguraPlaska* object) {
```
Comment (facku8) [inline lab1PO/FigureList.cpp:41]: Where are those methods used?

Comment (facku8) [reply lab1PO/FigureList.cpp:41]: And why they (setters) are exposed outside?

File: lab1PO/main.cpp [hunk 5]
```diff
@@ -1,89 +1,27 @@
...
 
 		switch (operation) {
 			case 'v':
-				cout << endl << *list.current << endl;
+				cout << endl << *list->getCurrent() << endl;
 				break;
 			case 'n':
-				i++;
-				updateList(&list, figures, operation, &i, listLength);
+				list->updateList(operation);
```
Comment (facku8) [inline lab1PO/main.cpp:18]: seperate function on list: addNewElement()

File: lab1PO/main.cpp [hunk 6]
```diff
@@ -1,89 +1,27 @@
...
 				break;
 			case 'p':
-				updateList(&list, figures, operation, &i, listLength);
+				list->updateList(operation);
```
Comment (facku8) [inline lab1PO/main.cpp:21]: separatce function on list getPreviousElement or whatever 'p' means

File: lab1PO/FigureList.cpp (hunk 2 above)
Comment (facku8) [reply lab1PO/FigureList.cpp:61]: Or it should rather be part of executeProgram. When you gather all input (circle vs rect vs square) you could just call relevant function of figuresList object.

Thanks to this, you will have a separation of input and features list logic.

File: lab1PO/FigureList.cpp [hunk 7]
```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include "FigureList.h"
+#include "Kolo.h"
+#include "Prostokat.h"
+#include "Trojkat.h"
+using namespace std;
+
+
+FigureList::FigureList(double a, double b, double c, double r) {
```
Comment (facku8) [inline lab1PO/FigureList.cpp:9]: This constructor seems to be not natural. IMHO, there should be on argument with an array called initialFigures.

**Comment by Nowak0** [comment]
111

**Comment by facku8** [comment]
test 123
